ProcessLib/Output.cpp

```diff
@@ -134,46 +134,39 @@ void Output::addProcess(ProcessLib::Process const& process,
                                  std::forward_as_tuple(filename));
 }
 
-Output::SingleProcessData Output::findSingleProcessData(
-    Process const& process, const unsigned process_id) const
-{
-    if (process.isMonolithicSchemeUsed())
-    {
-        auto spd_it = _single_process_data.find(&process);
-        if (spd_it == _single_process_data.end()) {
-            OGS_FATAL("The given process is not contained in the output"
-                      " configuration. Aborting.");
-        }
-        return spd_it->second;
-    }
-
-    auto spd_range = _single_process_data.equal_range(&process);
-    unsigned counter = 0;
-    for (auto spd_it=spd_range.first; spd_it!=spd_range.second; ++spd_it)
-    {
-        if(counter == process_id)
-        {
-            return spd_it->second;
-        }
-        counter++;
-    }
-
-    OGS_FATAL("The given process is not contained in the output"
-                      " configuration. Aborting.");
-}
-
-
 void Output::doOutputAlways(Process const& process,
                             const int process_id,
                             ProcessOutput const& process_output,
                             unsigned timestep,
                             const double t,
                             GlobalVector const& x)
 {
+    const bool make_output =
+        !(process_id < static_cast<int>(_single_process_data.size()) - 1 &&
+          !(process.isMonolithicSchemeUsed()));
```

Comment:
I think the expression would be more readable if it was rephrased as:
```
process_id == static_cast<int>(_single_process_data.size()) - 1 
         || process.isMonolithicSchemeUsed();
```

Btw.: Is there a special reason why the last process writes output (i.e. `process_id == static_cast<int>(_single_process_data.size()) - 1`), and not the first one?

Reply:
Changed. A comment was added as the reason as
```
    // For the staggered scheme for the coupling, only the last process, which
    // gives the latest solution within a coupling loop, is allowed to make
    // output.
```

Reply:
OK, thanks.